src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,189 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import java.util.Random;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int MIN_AGE = 18;
+    private static final int BOUND_NUMBER = 10;
+    private static StorageDao storageDao;
+    private static Random random;
+    private static RegistrationService registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        storageDao = new StorageDaoImpl();
+        random = new Random();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_existingUser_throwsException() {
+        User user1 = new User("Recruit", "password", MIN_AGE);
+        storageDao.add(user1);
+
+        User user2 = new User("Recruit", "KingBoss", MIN_AGE);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    public void register_loginLength_lessThan6_throwsException() {
+        User user = new User("PON", "MyCode", MIN_AGE);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_loginLength_5_registrationFailed() {
+        User user = new User("derek", "derekD12", MIN_AGE);
+
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void register_loginLength_6_registrationSuccessful() {
+        User user = new User("Peters", "passCodes", MIN_AGE);
+
+        assertDoesNotThrow(() -> registrationService.register(user));
```

Comment:
```suggestion
        User actual = registrationService.register(user);
        assertEquals(user, actual);
```

The same for other OK cases. BTW you can write only one valid test case